### ISSUE TITLE
queue:work not running, cannot fetch mails

### ISSUE THREAD

**Issue by NormanBaechtold** [freescout-helpdesk/freescout#1903]
I have successfully setup freescout, connected an e-mail account and registered our users. I can send mails to other through the freescout interface, but cannot receive any. To solve this issue, I followed the troubleshooting guidelines [here ](https://github.com/freescout-helpdesk/freescout/wiki/Fetching-Emails#troubleshooting-fetching-issues). In System Status, freescout:fetch-emails operates as it should. However, queue:work is apparently not working: `Last run: ? Last successful run: ? Try to clear cache  to force command to start.`. 

Clearing the cache is not working. I don't get any errors in the fetching logs. They're only saying fetched 0 new mails". 

Any ideas what to do?

**Comment by mywebhero** [comment]
Freescout only fetch new unread email into the system. Mark any email and see if it starts to fetch email?
Also, did you setup cron job to trigger the fetching?

**Comment by NormanBaechtold** [comment]
Yes, the e-mails are unread. I setup cron according to the troubleshooting. The freescout:fetch-emails is setup as written in the troubleshooting. For the initial setup of freescout I used the docker-image, so I think that the schedule:run job is setup correctly as well.

**Comment by mywebhero** [comment]
If your cron is properly configured, the cron should work. I'm not sure if the screenshot helps me understand the issue. 

Will clicking the link in the cron command section help to trigger the email fetching?

**Comment by NormanBaechtold** [comment]
Clicking the link opens a new tab showing three lines "running scheduled command: [...]". Apart from that, nothing changes. The queue:work still has a "?" written in system status and there are no mails in the inbox.

**Comment by mywebhero** [comment]
It appears that you have two problems here. 

1. Your IMAP connection does not appear to be configured with the correct credentials. 
2. Your cron jobs are incorrectly configured. 

Are you hosting Freescout on a VPS or in the cloud? Or do you prefer shared hosting? To me, it does not appear to be the complete path. On some shared hosting accounts, the full path to the PHP executable (for example, /usr/local/bin/php-8.0) may be required.

**Comment by NormanBaechtold** [comment]
Okay, we will check the IMAP again. Can you tell what is wrong with the cron configuration? 

We are hosting Freescout on Debian Bullseye on a VM.

**Comment by mywebhero** [comment]
Sorry, I'm not good in server & VM. But my wild guess is your cross is calling the wrong folder?
My cron command is call a folder like this "php /home/[username]/[helpdesk URL]/artisan"

Yours is just www/html/artisan

I'm not sure if this is the full path the cron can locate. I hide the actual path for privacy, but you can tell the structure is different. Worth to try.

**Comment by NormanBaechtold** [comment]
That's definitely a good hint, thanks. I will look into it!

**Comment by NormanBaechtold** [comment]
So, after the week-end I have tried to fix the fetching-issue by looking into the discussed points above. My insights are: 

- Although the error says so, the issue can't be that I have set the wrong mailbox credentials, since I can send mails to my personal mailbox without any trouble. This accounts both for admin/system mails and mailbox mails. 
- Because I didn't know what could possibly be wrong with the cronjob, I decided to build the application from scratch, using the latest docker image (tiredofit/freescout:latest). I built the containers using `docker-compose up -d`. However, nothing changed. I can send mails, but not receive them. 

I'm quite sure the issue is rooted in the cronjob, but the question is: Where? Two things have come up in this regard:

- The non-docker installation guide says that schedule:run must be created for the user "www-data". However, when using `crontab -u www-data -e` inside the freescout-app docker-container, I get the error "no user www-data found". 
- A cronjob `schedule:run` is setup in the root crontab of the container (access via cronatb -e). It goes like this: 
  `***** sudo -u nginx TZ=Europe/Berlin LD_Preload=usr/lib/preloadable_libiconv.so php /www/html/artisan schedule:run >> /dev/null 2>&1`


So maybe the issue lies in the fact that there is no user www-data and no user-specific cronjob, respectively? If anyone agrees with this, then my questions are: 

1. How do I set up a new cronjob for user www-data. As I said, the command `crontab -u www-data -e` throws an error. 
2. Is the path /www/html/artisan correct? Since I run the app inside a docker-container, I do not see any other path than the standard one.
